src/configuration.rs

```diff
@@ -534,6 +536,52 @@ pub struct Extension {
     pub endpoint: String,
     // Deny/Allow request when faced with an irrecoverable failure.
     pub failure_mode: FailureMode,
+    #[serde(default)]
+    pub timeout: Timeout,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Timeout(pub Duration);
+impl Default for Timeout {
+    fn default() -> Self {
+        Timeout(Duration::from_millis(20))
+    }
+}
+
+impl<'de> Deserialize<'de> for Timeout {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(TimeoutVisitor)
+    }
+}
+
+struct TimeoutVisitor;
+impl<'de> Visitor<'de> for TimeoutVisitor {
+    type Value = Timeout;
+
+    fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+        formatter.write_str("DurationString -> Sign? Number Unit String? Sign -> '-' Number -> Digit+ ('.' Digit+)? Digit -> '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' Unit -> 'h' | 'm' | 's' | 'ms' | 'us' | 'ns' String -> DurationString")
+    }
+
+    fn visit_str<E>(self, string: &str) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        self.visit_string(String::from(string))
+    }
+
+    fn visit_string<E>(self, string: String) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        match duration(Arc::new(string)) {
+            Ok(Value::Duration(duration)) => Ok(Timeout(duration.to_std().unwrap())),
```

Comment:
I think this could be an issue here, if I understand this right a "`CelDuration`" can be negative, a std can't if I'm not mistaken. So we expose it as if it could, then would panic if one would provide a negative one, which would be nonsensible in any case tho...

Reply:
you are correct, this would panic on a negative as `OutOfRangeError(())`